### PR TITLE
Formalize Notational Conventions

### PR DESCRIPTION
**Pull request by atlaskerr**
This is an administrative PR that adds proper grammar to notational conventions.

This branch should be ready to merge later this week.

### CODE REVIEW

**Comment by dmcgowan** [comment]
LGTM

Did you want to wait until later to get this in or just do another pass PR later based on changes. We could certainly try enforce this with newer PRs.

File: spec.md [hunk 1]
```diff
@@ -262,22 +262,22 @@ A digest can be verified by independently calculating `D` and comparing it with
...
 This is most important when fetching by a digest.
-To ensure security, the content should be verified against the digest used to fetch the content.
-At times, the returned digest may differ from that used to initiate a request.
+To ensure security, the content SHOULD be verified against the digest used to fetch the content.
+At times, the returned digest MAY differ from that used to initiate a request.
 Such digests are considered to be from different _domains_, meaning they have different values for _algorithm_.
-In such a case, the client may choose to verify the digests in both domains or ignore the server's digest.
-To maintain security, the client _must_ always verify the content against the _digest_ used to fetch the content.
+In such a case, the client MAY choose to verify the digests in both domains or ignore the server's digest.
+To maintain security, the client _MUST_ always verify the content against the _digest_ used to fetch the content.
```
Comment (mikebrow) [inline spec.md]: `/s/_MUST_/MUST/`

Italicizing had merit when it was lower case.. but we SHOULD be consistent with the legal spec terms.

File: spec.md [hunk 2]
```diff
@@ -543,9 +543,9 @@ Content-Type: application/octet-stream
 <Last Layer Chunk Binary Data>
 ```
 
-Optionally, if all chunks have already been uploaded, a `PUT` request with a `digest` parameter and zero-length body may be sent to complete and validate the upload.
-Multiple "digest" parameters may be provided with different digests.
-The server may verify none or all of them but _must_ notify the client if the content is rejected.
+Optionally, if all chunks have already been uploaded, a `PUT` request with a `digest` parameter and zero-length body MAY be sent to complete and validate the upload.
+Multiple "digest" parameters MAY be provided with different digests.
+The server MAY verify none or all of them but _MUST_ notify the client if the content is rejected.
```
Comment (mikebrow) [inline spec.md]: ditto

File: spec.md [hunk 3]
```diff
@@ -870,19 +870,19 @@ To get the next result set, a client would issue the request as follows, using t
...
-A delete may be issued with the following request format:
+An image MAY be deleted from the registry via its `name` and `reference`.
+A delete MAY be issued with the following request format:
 
 ```HTTP
     DELETE /v2/<name>/manifests/<reference>
 ```
 
-For deletes, `reference` *must* be a digest or the delete will fail.
+For deletes, `reference` *MUST* be a digest or the delete will fail.
```
Comment (mikebrow) [inline spec.md]: ditto

File: spec.md [hunk 4]
```diff
@@ -726,30 +726,30 @@ Content-Type: application/json
 }
 ```
 
-Note that catalog operations are optional for registry implementations.
+Note that catalog operations are OPTIONAL for registry implementations.
 Contents of the response are specific to the registry implementation.
-Some registries may opt to provide a full catalog output, limit it based on the user's access level or omit upstream results, if providing mirroring functionality.
-Subsequently, the presence of a repository in the catalog listing only means that the registry *may* provide access to the repository at the time of the request.
+Some registries MAY opt to provide a full catalog output, limit it based on the user's access level or omit upstream results, if providing mirroring functionality.
+Subsequently, the presence of a repository in the catalog listing only means that the registry *MAY* provide access to the repository at the time of the request.
```
Comment (mikebrow) [inline spec.md]: ditto

File: spec.md [hunk 5]
```diff
@@ -438,8 +438,8 @@ Content-Length: 0
 The rest of the upload process can be carried out with the returned url, called the "Upload URL" from the `Location` header.
 All responses to the upload url, whether sending data or getting status, will be in this format.
 
-Though the URI format (`/v2/<name>/blobs/uploads/<uuid>`) for the `Location` header is specified, clients should treat it as an opaque url and should never try to assemble it.
-While the `uuid` parameter may be an actual UUID, this proposal imposes no constraints on the format and clients should never impose any.
+Though the URI format (`/v2/<name>/blobs/uploads/<uuid>`) for the `Location` header is specified, clients SHOULD treat it as an opaque url and SHOULD never try to assemble it.
+While the `uuid` parameter MAY be an actual UUID, this proposal imposes no constraints on the format and clients SHOULD never impose any.
 
 Header `Blob-Upload-UUID` *OPTIONAL*: If clients need to correlate local upload state with remote upload state, largely for resumable uploads.
```
Comment (mikebrow) [inline spec.md]: ditto...

Comment (atlaskerr) [reply spec.md]: On this one, should `SHOULD` and `MAY` be lowercase? Or just `SHOULD`?

Comment (atlaskerr) [reply spec.md]: Ohhhh wait you want me to remove the italic and bold stuff from notations! Sorry I misunderstood.

Comment (mikebrow) [reply spec.md]: On this one the issue is `*OPTIONAL*` should just be OPTIONAL

Comment (atlaskerr) [reply spec.md]: Ah it was staring me right in the face.  Done.

**Pull request merged**